lib/tokenizers.js

```diff
@@ -17,6 +17,10 @@ let AUTO_ROTATION_CHECKER = null;
 bedrock.events.on('bedrock-mongodb.ready', async () => {
   await database.openCollections(['tokenizer-tokenizer']);
 
+  // the created indexes of 'tokenizer.id', 'tokenizer.state' and
+  // 'tokenizer.current' ensure that all queries are indexed properly. No
+  // additional indexes should be created here unless additional queries need
+  // to be added.
```

Comment:
This is general advice that should *always* be true for all of our modules. IMO, that means this comment should appear in all of our modules -- or none of them. I'd vote for the latter. It's simply best practice to follow this advice -- we don't need to document it as a comment. To me, it is similar to making other basic value statements in comments in code like this:

```js
// always use short, but meaningful variable names
const isSettled = checkIfSettled();
```

Reply:
Well @mattcollier made a good point that someone else who may not be familiar with this code, might come in and change things without the knowledge of making sure all the indexes are working. 

So providing as much comment details to unfamiliar users might be good, otherwise if changes are made and audits are not done for those changes, it defeats the purpose of the auditing I am doing. It will later have to be done again. 

I get that its a lot of work adding these comments, but if I didn't see these comments, it would be very easy for me to just add an additional query to a search that could cause a collection scan instead of an index scan. If I was not knowledgable about mongo then it's possible that I would not even know that I am breaking or causing a large slowdown in our system. 

This could be an issue in 6 months or 6 years because we could have lots of different people touch this code in the future. I think more details about this is better than no details to prevent future indexing problems. However, if it is preferred to remove the comments, I get it, but I think it will be hard for others to know that this code was audited and all queries were properly indexed.

Reply:
> Well @mattcollier made a good point that someone else who may not be familiar with this code, might come in and change things without the knowledge of making sure all the indexes are working.

They may still do that and simply ignore the comment. Test coverage and failed tests should be our measure for ensuring these things don't regress.

Reply:
We should clearly document the specific threat we're worried about. Someone changing an existing query should be caught by tests. Someone adding a new query / new function would not be. However, I suspect that a comment in a function you're not working on is not going to be a very good protection against the latter -- and it may only marginally help with the former.

Tests are the best mechanism for protecting against the former threat -- and I don't know what we want to do about the latter other than requiring better test coverage for new functions. We don't have a test / other kind of forcing function to make someone write an index checking test. It's an interesting problem to solve, but I think having these comments won't solve it -- I think it will just increase the maintenance burden.

Reply:
@dlongley Understood. Yes the tests I am writing will catch the query problems. I will go ahead and remove the comments and add tests for any repeated queries used by another function to ensure better test coverage.